cmake/modules/AddSwift.cmake

```diff
@@ -546,8 +548,81 @@ macro(add_swift_lib_subdirectory name)
   add_llvm_subdirectory(SWIFT LIB ${name})
 endmacro()
 
+function(_link_host_swift_system_libs executable)
+  target_link_directories(${executable} PRIVATE
+    ${CMAKE_OSX_SYSROOT}/usr/lib/swift)
+  set_target_properties(${executable} PROPERTIES
+    BUILD_WITH_INSTALL_RPATH YES
```

Comment:
I would rather that you add these as interface libraries of libswift. That will ensure everything that links against libswift will get it and you can remove all of this code.